### PR TITLE
Update errorprone

### PR DESCRIPTION
**Pull request by shemnon**
* Update errorprone from 2.3.1 to 2.3.2
* Update Jenkinsfile so that CI will use Java 11
* Suppress these new rules:
  * EqualsGetClass
  * ImmutableEnumChecker
  * UnnecessaryParentheses
* Change code to conform to these new rules:
  * BadImport
  * BadInstanceof
  * InconsistentHashCode
  * LockNotBeforeTry
  * MathAbsoluteRandom
  * ModifiedButNotUsed
  * UndefinedEquals

### CODE REVIEW

File: ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java [hunk 1]
```diff
@@ -326,17 +326,18 @@ public void logsShouldBeFlaggedAsRemovedWhenBlockIsNotInCanonicalChain() {
     final Block fork = gen.block(options);
     final List<TransactionReceipt> forkReceipts = gen.receipts(fork);
 
-    final List<Block> reorgedChain = new ArrayList<>(blocks.subList(0, forkBlock));
-    reorgedChain.add(fork);
-    final List<List<TransactionReceipt>> reorgedReceipts =
-        new ArrayList<>(blockReceipts.subList(0, forkBlock));
-    reorgedReceipts.add(forkReceipts);
+    // final List<Block> reorgedChain = new ArrayList<>(blocks.subList(0, forkBlock));
```
Comment (lucassaldanha) [inline ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java]: Did you mean to leave this snippet commented?

File: ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java [hunk 2]
```diff
@@ -326,17 +326,18 @@ public void logsShouldBeFlaggedAsRemovedWhenBlockIsNotInCanonicalChain() {
...
+    // reorgedChain.add(fork);
+    // final List<List<TransactionReceipt>> reorgedReceipts =
+    //     new ArrayList<>(blockReceipts.subList(0, forkBlock));
+    // reorgedReceipts.add(forkReceipts);
```
Comment (lucassaldanha) [inline ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java]: Same thing here about the comments.

Comment (shemnon) [reply ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java]: same

File: ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java (hunk 1 above)
Comment (shemnon) [reply ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/queries/BlockchainQueriesTest.java]: Placeholder to show it was being deleted and to check for side effects.  I'll delete it.

**Pull request merged**